nevergrad/functions/base.py

```diff
@@ -204,6 +203,59 @@ def update_leaderboard(identifier: str, loss: float, array: np.ndarray, verbose:
         pass  # better avoir bugs for this
 
 
+class ArrayExperimentFunction(ExperimentFunction):
+    """Combines a function and its parametrization for running experiments (see benchmark subpackage).
+    Extends ExperimentFunction, in the special case of an array, by allowing the creation of symmetries
+    of a single function. We can create ArrayExperimentFunction(callable, symmetry=i) for i in range(0, 2**d)
+    when the callable works on R^d.
+    Works only if there are no constraints.
+
+    Parameters
+    ----------
+    function: callable
+        the callable to convert
+    parametrization: Parameter
+        the parametrization of the function
+    symmetry: int
+        number parametrizing how we symmetrize the function.
+    """
+
+    def __init__(self, function: tp.Callable[..., tp.Loss], parametrization: p.Parameter, symmetry: int = 0) -> None:
+        """Adds a "symmetry" parameter, which allows the creation of many symmetries of a given function.
+
+        symmetry: an int, 0 by default.
+        if not zero, a symmetrization is applied to the input; each of the 2^d possible values
+        for symmetry % 2^d gives one different function.
+        Makes sense if and only if (1) the input is a single ndarray (2) the domains are symmetric."""
+        super().__init__(function, parametrization)
+        assert isinstance(parametrization, p.Array), f"{type(parametrization)} is not p.Array; {parametrization.parameters}."
+        assert (parametrization.bounds[0] is None) == (parametrization.bounds[1] is None)
+        assert len(parametrization._constraint_checkers) == 0
+        assert symmetry >= 0
+        assert symmetry < 2 ** self.dimension
+        self._inner_function = self._function
+        self._symmetry = symmetry
+        if self._symmetry != 0:
+            self._function = self.symmetrized_function
+        else:
+            self._function = function
+
+    def symmetrized_function(self, x: np.ndarray) -> tp.Loss:
+        assert isinstance(x, np.ndarray), "symmetry != 0 works only when the input is an array."
+        assert len(x.shape) == 1, "only one-dimensional arrays for now."
+        y = x
+        symmetry: int = self._symmetry  # type: ignore
+        for i in range(len(y)):
+            if symmetry % 2 == 1:
+                if self.parametrization.bounds[0] is not None and self.parametrization.bounds[1] is not None:  # type: ignore
+                    middle = (self.parametrization.bounds[0][i] + self.parametrization.bounds[1][i]) / 2.  # type: ignore
+                else:
+                    middle = 0.
+                y[i] = middle - x[i]  # We should rather symmetrize w.r.t the center of Parameter. TODO
+            symmetry = symmetry // 2
+        return self._inner_function(y)  # type: ignore
```

Comment:
once this is tested and works (and possibly merged) a better option of doing this is to precomute the transform that must be applied (it's basically  a x + b  if I am not forgetting anything, so we only need a and b

Reply:
Reimplemented that way.